source/server/admin/logs_handler.cc

```diff
@@ -51,64 +62,134 @@ Http::Code LogsHandler::handlerReopenLogs(absl::string_view, Http::ResponseHeade
       }
 
-      if (!logger_to_change) {
-        return false;
+      auto level_to_use = parseLevel(level);
+      if (!level_to_use.has_value()) {
+        return std::make_pair(false, "unknown logger level");
       }
 
-      logger_to_change->setLevel(static_cast<spdlog::level::level_enum>(level_to_use));
+      name_levels[name] = *level_to_use;
     }
+  } else {
+    // Change particular log level by name.
+    auto level_to_use = parseLevel(value);
+    if (!level_to_use.has_value()) {
+      return std::make_pair(false, "unknown logger level");
+    }
+
+    name_levels[key] = *level_to_use;
+  }
+
+  return changeLogLevels(name_levels);
+}
+
+void LogsHandler::changeAllLogLevels(spdlog::level::level_enum level) {
+  if (!Logger::Context::useFancyLogger()) {
+    ENVOY_LOG(debug, "change all log levels: level='{}'", spdlog::level::level_string_views[level]);
+    Logger::Registry::setLogLevel(level);
   } else {
     // Level setting with Fancy Logger.
-    spdlog::level::level_enum lv = static_cast<spdlog::level::level_enum>(level_to_use);
-    if (name == "level") {
-      FANCY_LOG(info, "change all log levels: level='{}'", level);
-      getFancyContext().setAllFancyLoggers(lv);
-    } else {
-      FANCY_LOG(info, "change log level: name='{}' level='{}'", name, level);
-      bool res = getFancyContext().setFancyLogger(name, lv);
-      return res;
+    FANCY_LOG(info, "change all log levels: level='{}'", spdlog::level::level_string_views[level]);
+    getFancyContext().setAllFancyLoggers(level);
+  }
+}
+
+std::pair<bool, std::string> LogsHandler::changeLogLevels(
+    const absl::flat_hash_map<absl::string_view, spdlog::level::level_enum>& changes) {
+  if (!Logger::Context::useFancyLogger()) {
+    std::vector<std::pair<Logger::Logger*, spdlog::level::level_enum>> loggers_to_change;
+    for (Logger::Logger& logger : Logger::Registry::loggers()) {
+      auto name_level_itr = changes.find(logger.name());
+      if (name_level_itr == changes.end()) {
+        continue;
+      }
+
+      loggers_to_change.emplace_back(std::make_pair(&logger, name_level_itr->second));
+    }
+
+    // Check if we have any invalid logger in changes.
+    if (loggers_to_change.size() != changes.size()) {
+      return std::make_pair(false, "unknown logger name");
+    }
+
+    for (auto& it : loggers_to_change) {
+      Logger::Logger* logger = it.first;
+      spdlog::level::level_enum level = it.second;
+
+      ENVOY_LOG(debug, "change log level: name='{}' level='{}'", logger->name(),
+                spdlog::level::level_string_views[level]);
+      logger->setLevel(level);
+    }
+  } else {
+    std::vector<std::pair<SpdLoggerSharedPtr, spdlog::level::level_enum>> loggers_to_change;
+    for (auto& it : changes) {
+      std::string name(it.first.data(), it.first.size());
+      spdlog::level::level_enum level = it.second;
+
+      auto logger = getFancyContext().getFancyLogEntry(name);
```

Comment:
nit you can inline the conversion to the string

but more importantly use an explicit type rather than `auto` here as it is not obvious from context what the type is.
```
SpdLogSharedPtr logger = getFancyContext().getFancyLogEntry(std::string(it.first));
```

Reply:
Also can you leave a TODO to change `getFancyLogger` to operate with string_view?